src/protocols/at-ps2/common_interface.c

```
@@ -55,4 +61,39 @@ uint8_t interface_parity_table[256] = {
     0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0,  // E
     1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1   // F
-};
\ No newline at end of file
+};
+
+void atps2_send_command(pio_engine_t* engine, uint8_t data_byte) {
+    uint16_t data_with_parity = (uint16_t)(data_byte + (interface_parity_table[data_byte] << 8));
+    pio_sm_put(engine->pio, engine->sm, data_with_parity);
+}
+
+bool atps2_setup_pio_engine(pio_engine_t* engine, uint data_pin, pio_irq_callback_t callback,
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'atps2_setup_pio_engine' of similar type ('int') are easily swapped by mistake

(bugprone-easily-swappable-parameters)